packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -480,6 +483,37 @@ const baseDescriptors: PropertyDescriptorMap = ArrayReduce.call(
 
 defineProperties(BaseLightningElement.prototype, baseDescriptors);
 
+// Per Component Constructor, track the corresponding Custom Element
+const ContextProviderMetaMap = new Map();
+
+function getCustomElement(Ctor: ComponentConstructor): HTMLElement {
+    if ((Ctor as any) === BaseLightningElement) {
+        throw new SyntaxError(`Invalid Constructor`);
```

Comment:
This error is not a `SyntaxError` per se since it doesn't have any bearing on the actual class syntax. I would rather convert the error to a `TypeError`.